CONTRIBUTING.md

```diff
@@ -1,29 +1,157 @@
+
+
+### Proposing a Change
+1. Open a new issue if you would like report a bug or suggest improvements.
+2. Please wait for core team members to comment on the thread. This lets us reach an agreement on your proposal before you put significant effort into it.
+
+### Prerequisites
+1. [NodeJS](https://nodejs.org/)
+    <br/>
+    Minimum version v8.0.0+
     ```bash
-    git clone https://github.com/coderplex/coderplex.git
+    To check node version
+    $ node -v
     ```
-1. `cd coderplex`
-1. `npm install`
-1. `npm run dev` to start local development server
-1. App opens at `localhost:3000`
+    Any lower version than mentioned above may results in this [error](https://github.com/coderplex/coderplex/issues/100).
+    > If you face problem updating your node then you might need a Node version manager tool. [Follow here]()
+
+2. [Yarn](https://yarnpkg.com)
+    <br/>
+    Minimum version v1.2.0+
+    <br/>
+    Installing instructions are at [official docs](https://yarnpkg.com/en/docs/install#windows-tab).
+    Use yarn over npm
+    > Our team's official policy (for now) is: We only use [Yarn](https://yarnpkg.com/en/docs/install) as our official Node package manager, and so we request you to use Yarn instead of npm and commit `yarn.lock` file.
+
+3. [Git](https://git-scm.com/download/linux)
+    <br/>
+    Familiarity with git is mandatory.
+
+### Running
+
+1. Clone repository
+    ```bash
+    $ git clone https://github.com/coderplex/coderplex.git
+    ```
+2. Move to the repository root folder
+    ```bash
+    $ cd coderplex
+    ```  
+3. Install dependencies
+    ```bash
+    $ yarn
+    ```
+4. Start the development server
+    ```bash
+    $ yarn dev
+    ```
+    App now opens at `localhost:3000` in your default browser.
+    > You may get this [error]() if any other app is already running the above port.
+
+### Sending a Pull Request
+
+*Working on your first Pull Request? You can learn how from this *free* series [How to Contribute to an Open Source Project on GitHub](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github)*
+
+#### Before submitting:
+1. Fork the [repository](https://github.com/coderplex/coderplex).
+2. From your fork, create a [branch](https://help.github.com/articles/creating-and-deleting-branches-within-your-repository/) and name it. eg. `typo-in-readme`
+3. If you’ve fixed a bug or added code that should be tested, add tests!
+4. Ensure that all test pass
+    ```bash
+    $ yarn test
+    ```
+5. Run code formatters
+    ```bash
+    $ yarn lint
+    ```
+6. Add and commit your code. Please give meaning full commit messages.
+7. Pull latest code from [upstream repository's ](https://help.github.com/articles/merging-an-upstream-repository-into-your-fork/)`master`, if in case anything new were merged while you were working on your fork.
+8. Push the code to your fork.
+9. Raise the pull request from your created branch to `master` branch of coderplex.
```

Comment:
shoud we change this to `develop`. and explain a little about out branching system?

Reply:
ping!

Reply:
Change what to `develop`?

Reply:
Yes we will be changing default branch as `devlop` (I made typo while creating this will modify soon to develop). So update `master` to `develop` in guide and mention why we are doing so.